pkg/deploy/api/v1/conversion.go

```diff
@@ -172,46 +37,17 @@ func convert_api_DeploymentTriggerImageChangeParams_To_v1_DeploymentTriggerImage
 	if err := s.Convert(&in.From, &out.From, 0); err != nil {
 		return err
 	}
-	switch in.From.Kind {
-	case "ImageStream":
+	if out.From.Kind != "ImageStreamTag" {
 		out.From.Kind = "ImageStreamTag"
-		tag := in.Tag
-		if len(tag) == 0 {
-			tag = imageapi.DefaultImageTag
-		}
-		out.From.Name = fmt.Sprintf("%s:%s", in.From.Name, tag)
-	}
-	return nil
-}
-
-func convert_v1_DeploymentCauseImageTrigger_To_api_DeploymentCauseImageTrigger(in *DeploymentCauseImageTrigger, out *newer.DeploymentCauseImageTrigger, s conversion.Scope) error {
-	switch in.From.Kind {
-	case "DockerImage":
-		ref, err := imageapi.ParseDockerImageReference(in.From.Name)
-		if err != nil {
-			return err
-		}
-		out.Tag = ref.Tag
-		ref.Tag, ref.ID = "", ""
-		out.RepositoryName = ref.Minimal().String()
-	}
-	return nil
-}
-
-func convert_api_DeploymentCauseImageTrigger_To_v1_DeploymentCauseImageTrigger(in *newer.DeploymentCauseImageTrigger, out *DeploymentCauseImageTrigger, s conversion.Scope) error {
-	if len(in.RepositoryName) != 0 {
-		ref, err := imageapi.ParseDockerImageReference(in.RepositoryName)
-		if err != nil {
-			return err
-		}
-		ref.Tag = in.Tag
-		out.From.Kind = "DockerImage"
-		out.From.Name = ref.String()
+		out.From.Name = imageapi.JoinImageStreamTag(out.From.Name, imageapi.DefaultImageTag)
 	}
 	return nil
 }
 
 func convert_v1_RollingDeploymentStrategyParams_To_api_RollingDeploymentStrategyParams(in *RollingDeploymentStrategyParams, out *newer.RollingDeploymentStrategyParams, s conversion.Scope) error {
+	if defaulting, found := s.DefaultingInterface(reflect.TypeOf(*in)); found {
```

Comment:
I would have expected defaulting to already have happened by the time we get here.  Is it not?